.github/workflows/push-docker-debug.yaml

```diff
@@ -1,18 +1,86 @@
-name: docker image
+name: publish docker image
 
 on:
   pull_request:
+    paths:
+      - docker/Dockerfile*
+      - scripts/install_*.sh
+      - .github/workflows/push-docker-*.yaml
     types:
       - labeled
+  merge_group:
+    paths:
+      - docker/Dockerfile*
+      - scripts/install_*.sh
+      - .github/workflows/push-docker-*.yaml
+  push:
+    branches:
+      - main
+  release:
+    types: [published]
+
+env:
+  DOCKER_REPOSITORY: "tigrisdata/tigris-debug"
+  QUAY_REPOSITORY: "quay.io/tigrisdata/tigris-debug"
 
 jobs:
-  debug:
-    if: ${{ github.event.label.name == 'debug image' }}
-    uses: ./.github/workflows/push-docker-image.yaml
-    secrets: inherit
-    with:
-      docker_repository: "tigrisdata/tigris-debug"
-      quay_repository: "quay.io/tigrisdata/tigris-debug"
-      file: docker/Dockerfile
-      platforms: amd64
-      event_name: push
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: Fetch tags
+        run: |
+          git fetch --prune --unshallow --tags
+
+      - name: Login to Docker Hub
+        id: login-docker-hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.GH_DOCKER_ACCESS_USER }}
+          password: ${{ secrets.GH_DOCKER_ACCESS_TOKEN }}
+
+      - name: Login to Quay.io
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_REGISTRY_USER }}
+          password: ${{ secrets.QUAY_REGISTRY_PASSWORD }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ${{ env.DOCKER_REPOSITORY }}
+            ${{ env.QUAY_REPOSITORY }}
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=raw,value=tigris-debug-{{version}}-{{major}}.{{minor}},enable=${{ github.event_name = 'release' }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and push debug images to DockerHub
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: docker/Dockerfile
+          platforms: linux/amd64,linux/arm64
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          build-args: ${{ inputs.build_args }}
```

Comment:
You need to add this in `.github/workflows/push-docker-image.yaml`.

In this file we only call `.github/workflows/push-docker-image.yaml` and pass build_args like this:
```
jobs:
  debug:
    if: ${{ github.event.label.name == 'debug image' }}
    uses: ./.github/workflows/push-docker-image.yaml
    secrets: inherit
    with:
      docker_repository: "tigrisdata/tigris-debug"
      quay_repository: "quay.io/tigrisdata/tigris-debug"
      file: docker/Dockerfile
      platforms: amd64
      event_name: push
      build_args: WITH_DEBUG=1
```

Reply:
Let me take a look. I admit that I haven't reviewed the new structure in place to see what has been standardized and deduplicated. Stay tuned!

Reply:
OK so on publish we seem to run `push-docker-image` with arguments like platforms and you want me to add this not as standalone build but rather part of the base `push-docker-image` and call it with the extra argument?

Reply:
Please review again, modified based on above.